src/lidar_processor.cpp

```diff
@@ -76,37 +76,38 @@ void LidarProcessor::raw_pc_callback(const sensor_msgs::msg::PointCloud2::Shared
 
   // *** Begin filter ***
   // Container for original & filtered data
-  pcl::PointCloud<pcl::PointXYZ>::Ptr cloud(new pcl::PointCloud<pcl::PointXYZ>);
+  pcl::PointCloud<pcl::PointXYZI>::Ptr cloud(new pcl::PointCloud<pcl::PointXYZI>);
 
   // Convert to PCL data type
   pcl::fromROSMsg(*msg, *cloud);
 
   // Perform the Passthrough filtering
-  pcl::PassThrough<pcl::PointXYZ> pass;
+  pcl::PassThrough<pcl::PointXYZI> pass;
   pass.setInputCloud(cloud);
   pass.setFilterFieldName("z");
   pass.setFilterLimits(0.0, 0.5);
```

Comment:
Could you add a line for intensity? I believe you just need to add line 87/88 but for "i"

Intensity is also a float between 0 and 1 so maybe cutting off all points below 0.3? [0.3,1.0]?